File: BookingApp/frontend/src/app/shared/models/ReservationPenalty.ts
import { Reservation } from "./Reservation";
import { QuickReservation } from "./reservations/QuickReservation";

export class ReservationPenalty{
    public id : Number;
    public reservationId:Number;
    public reservation : Reservation;

    constructor(id : Number, reservationId : Number){
        this.id = id;
        this.reservationId = reservationId;
    }
}

File: BookingApp/src/main/java/com/example/BookingApp/reservations/model/ReservationPenalty.java
package com.example.BookingApp.reservations.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.GenericGenerator;

import javax.persistence.*;

@Entity
@NoArgsConstructor
@Getter
@Setter
@AllArgsConstructor
@Table(name="reservationPenalties")
public class ReservationPenalty {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO, generator="seq")
    @GenericGenerator(name = "seq", strategy="increment")
    private Long id;
    @ManyToOne(fetch = FetchType.LAZY, cascade = CascadeType.DETACH)
    private Reservation reservation;
}
